src/base/bittorrent/torrentinfo.cpp

```diff
@@ -362,24 +362,29 @@ int BitTorrent::TorrentInfo::fileIndex(const QString& fileName) const
     return -1;
 }
 
-bool TorrentInfo::hasRootFolder() const
+QString TorrentInfo::rootFolder() const
 {
-    QString testRootFolder;
+    QString rootFolder;
     for (int i = 0; i < filesCount(); ++i) {
         const QString filePath = this->filePath(i);
         if (QDir::isAbsolutePath(filePath)) continue;
 
         const auto filePathElements = filePath.splitRef('/');
         // if at least one file has no root folder, no common root folder exists
-        if (filePathElements.count() <= 1) return false;
+        if (filePathElements.count() <= 1) return "";
```

Comment:
I've done some digging in Qt source, seems `QString()` can be [constructed](https://github.com/qt/qtbase/blob/5.11/src/corelib/tools/qstring.h#L1129) at compile-time, while `""` will be forced (or more likely) to be at [runtime](https://github.com/qt/qtbase/blob/5.11/src/corelib/tools/qstring.h#L693).

Reply:
I think it matters when declaring static constants. Here we create a temporary object at runtime, isn't it?
IMO, `return "";` is more readable.

Reply:
>Here we create a temporary object at runtime, isn't it?

Even so, QString() would still be a bit more efficient than "" when returning value.

>IMO, return ""; is more readable.

I didn't measure the performance difference, so I don't really insist on this.